### ISSUE TITLE
DynamoDB back end?

### ISSUE THREAD

**Issue by richardgavel** [outsideris/citizen#98]
Since Citizen is already AWS aware (using S3 as a backend for its packages), I wonder how hard it would be to use DynamoDB as a DB, borrowing from the implementation to use MongoDB. I know AWS does support managed MongoDB (via DocumentDB). However, that can be somewhat expensive, especially in a HA scenario, since the smallest size is a t3.medium. Especially since more than likely, the server is going to see very little activity. A serverless DB store might be a great option to have.

**Comment by richardgavel** [comment]
I did get DocumentDB to work (in my prior job, so don't have access to see details anymore). Are you sure you had the necessary security groups set up to allow port 27017 to communicate from citizen server to documentdb?

**Comment by joeykhashab** [comment]
interesting, yeah the security groups were fine. I manually logged into the ec2 instance that citizen was running on and used a "nc" command to verify the connection. The nc command worked.

Were you using 0.4.0 version or the master version of citizen?

did you have to create a database in DocumentDB manually before citizen connects to it?

**Comment by richardgavel** [comment]
Yes, i created the database ahead of time. And this was a fairly old
version of citizen.

**Comment by atomixer** [comment]
@joeykhashab I don't think Documentdb will work with citizen because [Prisma only supports Mongodb version 4.2 and above](https://www.prisma.io/docs/reference/database-reference/supported-databases). Documentdb is currently at version 4.0 so unless Amazon adds support for the newer versions of Mongodb, documentdb isn't really an option. Prisma adding support for Mongo v4.0 also [seems unlikely](https://github.com/prisma/prisma/issues/13205).

I was able to hookup the citizen server (v6.0) with a Documentdb instance (v4.0) but I kept running into errors when I tried to version modules.

**Comment by richardgavel-ordinaryexperts** [comment]
Yeah, when I got it to work, it was a version of Citizen so old that it didn't even have the HCL2 support in there, so at best it was 0.3.3.

**Comment by richardgavel-ordinaryexperts** [comment]
DocumentDB now apparently supports the MongoDB 5.0 compatability: https://docs.aws.amazon.com/documentdb/latest/developerguide/compatibility.html
